pages/api/policies/[policyId]/trustchains/index.ts

```diff
@@ -7,12 +7,16 @@ import ensureRole from 'src/middleware/ensureRole'
 import { Role } from 'src/config/guardianTags'
 import exceptionFilter from 'src/middleware/exceptionFilter'
 import onlyGet from 'src/middleware/onlyGet'
+import config from 'src/config'
+
+const isProtected = !config.publicTrustChainAccess
+const emptyHandler = (handler) => (req, res) => handler(req, res)
 
 export default prepare(
 	exceptionFilter,
 	onlyGet,
-	withHmac,
 	useGuardianContext,
-	withAuthentication,
-	ensureRole(Role.STANDARD_REGISTRY)
+	isProtected ? withHmac : emptyHandler,
```

Comment:
This is nonsense, split into a separate method above prepare.

```js
const generateMiddleware = () => {

  const base = [middlewares]

  if (!config.publicTrustChainAccess) {
    return [
       ...base
       (extras)
    ]
  }
  
  return base

  }

export default prepare(
  ...generateMiddleware()
)
```

Reply:
Thank you ser